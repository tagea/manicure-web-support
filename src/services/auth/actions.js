import {
  AUTHENTICATED,
  AUTHENTICATION_ERROR,
  LOGOUT,
  LOGOUT_ERROR
} from '../constants/ActionTypes';
import { getAuthToken, sendAuthToken, stateLocalStorage, removeLocalStorage } from '../api';

export const signInAction = (username, password) => async (dispatch) => {
  try {
    removeLocalStorage();
    const token = await getAuthToken(username, password);
    dispatch({ type: AUTHENTICATED });
    stateLocalStorage(token); // set
  } catch (error) {
    dispatch({
      type: AUTHENTICATION_ERROR,
      payload: error.message
    });
  }
};

export const signOutAction = () => async (dispatch) => {
  try {
    await sendAuthToken();
    removeLocalStorage();
    dispatch({ type: LOGOUT });
  } catch (error) {
    removeLocalStorage();
    dispatch({
      type: LOGOUT_ERROR,
      payload: error.message
    });
  }
};

// // note for future: add token check // //
export const stayInAction = () => (dispatch) => {
  try {
    const token = stateLocalStorage();
    if (token) {
      dispatch({ type: AUTHENTICATED });
    }
  } catch (error) {
    dispatch({
      type: AUTHENTICATION_ERROR,
      payload: error.message
    });
  }
};
