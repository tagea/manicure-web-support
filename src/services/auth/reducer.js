import {
  AUTHENTICATED,
  AUTHENTICATION_ERROR,
  UNAUTHENTICATION,
  LOGOUT,
  LOGOUT_ERROR
} from '../constants/ActionTypes';

const initState = {
  authenticated: false,
  error: false,
  status: null
};

export default function auth(state = {}, action) {
  switch (action.type) {
    case AUTHENTICATED:
      return { ...initState, authenticated: true, error: false };
    case AUTHENTICATION_ERROR:
      return { ...initState, error: true, status: action.payload };
    case UNAUTHENTICATION:
      return { ...initState, authenticated: false };
    case LOGOUT:
      return { ...initState, authenticated: false };
    case LOGOUT_ERROR:
      return { ...initState, status: action.payload };
    default:
      return state;
  }
}
