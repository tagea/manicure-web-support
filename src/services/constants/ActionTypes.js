export const AUTHENTICATED = 'AUTHENTICATED';
export const AUTHENTICATION_ERROR = 'AUTHENTICATION_ERROR';
export const UNAUTHENTICATION = 'UNAUTHENTICATION';
export const LOGOUT = 'LOGOUT';
export const LOGOUT_ERROR = 'LOGOUT_ERROR';
export const FETCH_SALON_SUCCESS = 'FETCH_SALON_SUCCESS';
export const FETCH_SALON_WAITING = 'FETCH_SALON_WAITING';
export const FETCH_SALON_FAILURE = 'FETCH_SALON_FAILURE';
export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
export const FETCH_USER_WAITING = 'FETCH_USER_WAITING';
