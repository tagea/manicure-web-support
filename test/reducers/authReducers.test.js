import auth from '../../src/services/auth/reducer';
import * as types from '../../src/services/constants/ActionTypes';

it('should return the initial state for AUTHENTICATED', () => {
  expect(auth(undefined, { type: types.AUTHENTICATED })).toMatchSnapshot();
});

it('should handle AUTHENTICATED', () => {
  expect(auth({}, {
    type: types.AUTHENTICATED,
    authenticated: true,
    error: false
  })).toEqual({
    authenticated: true,
    error: false,
    status: null
  });
});

it('should return the initial state for AUTHENTICATION_ERROR', () => {
  expect(auth(undefined, { type: types.AUTHENTICATION_ERROR })).toMatchSnapshot();
});

it('should return the initial state for LOGOUT', () => {
  expect(auth(undefined, { type: types.LOGOUT })).toMatchSnapshot();
});

it('should handle LOGOUT', () => {
  expect(auth({}, {
    type: types.LOGOUT
  })).toEqual({
    authenticated: false,
    error: false,
    status: null
  });
});

it('should return the initial state for LOGOUT_ERROR', () => {
  expect(auth(undefined, { type: types.LOGOUT_ERROR })).toMatchSnapshot();
});
