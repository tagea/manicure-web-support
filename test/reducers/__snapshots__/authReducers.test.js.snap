// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the initial state for AUTHENTICATED 1`] = `
Object {
  "authenticated": true,
  "error": false,
  "status": null,
}
`;

exports[`should return the initial state for AUTHENTICATION_ERROR 1`] = `
Object {
  "authenticated": false,
  "error": true,
  "status": undefined,
}
`;

exports[`should return the initial state for LOGOUT 1`] = `
Object {
  "authenticated": false,
  "error": false,
  "status": null,
}
`;

exports[`should return the initial state for LOGOUT_ERROR 1`] = `
Object {
  "authenticated": false,
  "error": false,
  "status": undefined,
}
`;
